src/Loader.php

```diff
@@ -157,6 +159,42 @@ public static function add_blocks(): void
         new Blocks\GuestBook();//NOSONAR
     }
 
+    /**
+     * Load editorIndex file and add core Query Loop Block for Beta Blocks.
+     */
+    public static function add_beta_blocks(): void
+    {
+        if (!BetaBlocks::is_active()) {
```

Comment:
I think we can move this statement to [here](https://github.com/greenpeace/planet4-master-theme/pull/2137/files#diff-1314d7908e6167b9fbe3d52c6f0c3cadf76637c75eca9c9c36d66106c579edd3R62). And we'll avoid to invoke unnecessary this function.

Reply:
I think that here we should also check that the blocks plugin is active, by adding `!Planet4Blocks::is_active()`

Reply:
> I think that here we should also check that the blocks plugin is active, by adding `!Planet4Blocks::is_active()` (like we do in the `add_blocks` function)

I am not sure, maybe @comzeradd can shed more light on this, but my understanding was this implementation was to exist despite us having to move the _Blocks_ to the theme folder, so the _Beta Blocks_ should be implemented even without the Planet4Blocks feature

Reply:
For now I think it's fine to just check for Beta blocks. Since this feature flag only exists on dev sites there is no danger for someone accidentally enabling this block in production.

Once we are ready to actual get this block live then we can substitute the Beta check with the `Planet4Blocks` feature flag.

Reply:
Maybe we can directly add this code to the `add_blocks` function? Instead of creating a new `add_beta_blocks`? 🤔

Reply:
Also, we already enqueue this script [in the MasterBlocks file](https://github.com/greenpeace/planet4-master-theme/blob/1200cf93530ee9a9e0a85569644ac74617971d2f/src/MasterBlocks.php#L51), should we remove that then if we add it here?

Reply:
Hey @mleray, thanks for the feedback.
So just a few points to see what you think. The reason for creating the `add_beta_blocks` function is so that in the future if we have to change things we can just isolate that particular code and integrate it into the `add_blocks`, depending on the route we choose. Also, it is easier to manage the script for the Beta blocks since we are enqueueing just that script while the `MasterBlock` file does a few other things.

Reply:
But won't we be loading the file twice then? 🤔

Reply:
I think the file is only loaded when the `Planet4Blocks` feature is enabled, which by default is not since that will throw an error because of the GuestBook block

Reply:
Actually I think most of our sites have it enabled, no? Doesn't it mean that our blocks plugin is enabled?

Reply:
Let me double check again.

Reply:
My bad, just realised that. I will have the code updated asap